### ISSUE TITLE
Questions on documentation & language service

### ISSUE THREAD

**Issue by zackarychapple** [SAP/angular-metaui#7]
## I'm submitting a...

<pre><code>
[ ] Bug report 
[ ] Feature request
[X] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

- oss language service is not public
- JAR source is not public

## Expected behavior

- When using IntelliJ syntax highlighting does not operate as expected. In the youtube video it appears there is a language service that is available. 
- How are new capabilities for operations and fields get added when the source to the parser is not available?

**Comment by fkolar** [comment]
Hi Zack, 

I have already replied to you in separate thread I hope we will get a chance for a chat.  But to sumarize: 

1.  To support our OSS grammer I wrote while ago a plugin for the IntellijIIDEA/WebStorm, where I created some version of support for this new language that at least understand the structure and does some highlighed. Its not perfect where I want to be, but good for now

2. Besides I might be talking to you I will probably update the https://github.com/SAP/angular-metaui/blob/master/lib/meta-ui-parser.md with more details, where I describe what is inside the JAR adn how ti works. 

But dont want to spent too much time with as I woudl rather focus on writing brand new AST parser that could work offline as well as in JIT mode. Which I already started in separate branch, you can some some grammer there as well.

**Comment by fkolar** [comment]
Reopening, as I want to have future issue to push docs updates... 

When you probably watch a demo it migh look straight forward until you try to integrate it into existing project where you get stack. This week or next week I will try to put together some get-started guide with some detail how things are connected. 

**For us is important to know:**

- How to integrate this into existing application or library
- Once its integrated what I can do and maybe explain some basic syntax tp get started
- Later on how to start extending system rules 

Later on once everything is document it will be easier either for me or for someone to quickly put together some schematics which automate all this.

**Comment by zackarychapple** [comment]
When it comes to integration into an existing application I think there are a couple aspects we should consider. 

- Is the integration only for forms / some forms 
- Is the integration going to include layout
- Is the integration going to take over the entire application

It would also be helpful to revisit how domain models are created in OSS.  For example it may be helpful to have them generated via a schematic from a TS class, or possibly be usable as the TS class itself. This however gets back into our discussion on the values and ease of use of OSS vs TS. 

Extending rules is a very exciting topic. I look forward to hearing more about that.

**Comment by fkolar** [comment]
I agree! I will start with the docs so others understand how to and why so they can have even more feedback adn ideas how to improve cuz I need more brains to this;-)

Lets start to think how to maximize use of TS, CLI, Schematics, anotations/decorators and as we discussed really maybe slowly move away frrom OSS because here we would gain:

- Having everything close to domain object and actual application
- Better syntax/type checking
- Easier to start as developers do not really need to learn anything new from what they already know
- Built in syntax highlighting and lang support without maintaining another set of plugins for different IDE

At last sometimes this or next week create new REPO that will focus only on MetaUI. This repo will stay and will be always seen as more full-stack with compoennts, where new REPO, should be componentless once we finish it.

**Comment by fkolar** [comment]
Started docs branch where I am dumping my notes

https://github.com/SAP/angular-metaui/tree/getStarted/packages/metaui/docs

So you can keep checking here before it will be merged to master. I will correct the docs and validate everything at once.

**Comment by tanmay-joshi-tfs** [comment]
Hi @fkolar, Shall we wait for meta-ui to become completely independent of PrimeNg before we can adopt it? Just curious because this dependency is mentioned under limitations. 
P.S.: I know this is not related to the issue raised here, but I didn't want to bother you with another issue for a question as trivial as this :)

**Comment by fkolar** [comment]
Hi Tammy , Meta ui is still under development and there is new repo that we created which has this feature to be library agnostic..

https://github.com/ngx-meta/rules
